pkg/apis/coordination/validation/validation.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/coordination"
+)
+
+// ValidateLease validates a Lease.
+func ValidateLease(lease *coordination.Lease) field.ErrorList {
+	allErrs := ValidateLeaseSpec(&lease.Spec, field.NewPath("spec"))
+	return allErrs
+}
+
+// ValidateLeaseUpdate validates an update of Lease object.
+func ValidateLeaseUpdate(lease, oldLease *coordination.Lease) field.ErrorList {
+	allErrs := ValidateLease(lease)
+	allErrs = append(allErrs, ValidateLeaseSpecUpdate(&lease.Spec, &oldLease.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+// ValidateLeaseSpec validates spec of Lease.
+func ValidateLeaseSpec(spec *coordination.LeaseSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if spec.LeaseDurationSeconds != nil && *spec.LeaseDurationSeconds <= 0 {
+		fld := fldPath.Child("leaseDurationSeconds")
+		allErrs = append(allErrs, field.Invalid(fld, spec.LeaseDurationSeconds, "must be greater than 0"))
+	}
+	if spec.LeaseTransitions != nil && *spec.LeaseTransitions < 0 {
+		fld := fldPath.Child("leaseTransitions")
+		allErrs = append(allErrs, field.Invalid(fld, spec.LeaseTransitions, "must to greater or equal than 0"))
+	}
+	return allErrs
+}
+
+// ValidateLeaseSpecUpdate validates an update of spec of Lease object.
+func ValidateLeaseSpecUpdate(spec, oldSpec *coordination.LeaseSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := ValidateLeaseSpec(spec, fldPath)
+
+	if spec.LeaseTransitions != nil && oldSpec.LeaseTransitions != nil {
```

Comment:
This makes it possible to set nil LeaseTransitions after setting it to a non-nil value. Maybe that's ok, and we allow clients to unset and reset, but that provides an easy way to get around this validation. It might be better to validate if just oldSpec.LeaseTransitions is non-nil, however.

Reply:
Also should we validate that the first value is always 0?